src/scipp/core/comparison.py

```diff
@@ -165,6 +166,12 @@ def identical(x: VariableLike,
         True if x and y have identical values, variances, dtypes, units,
         dims, shapes, coords, and masks. Else False.
     """
+    if isinstance(x, data_group.DataGroup):
```

Comment:
Missing a check for the type of `y`. Right now, the behaviour is inconsistent with other types. If called, e.g. with a variable and data array, it raises `TypeError`. But if called with a data group and data array, it raises `AttributeError` (from `y.keys()`). And if `y` is a dataset, it doesn't raise at all.